maven-scm-providers/maven-scm-providers-git/maven-scm-provider-jgit/src/main/java/org/apache/maven/scm/provider/git/jgit/command/checkin/JGitCheckInCommand.java

```diff
@@ -124,8 +127,11 @@ protected CheckInScmResult executeCheckInCommand(
                 UserInfo author = getAuthor(repo, git);
                 UserInfo committer = getCommitter(repo, git);
 
-                CommitCommand command = git.commit().setMessage(message).setAuthor(author.name, author.email);
-                command.setCommitter(committer.name, committer.email);
+                CommitCommand command = git.commit()
+                        .setMessage(message)
+                        .setAuthor(author.name, author.email)
+                        .setCommitter(committer.name, committer.email)
+                        .setSign(parameters.getBoolean(CommandParameter.SCM_COMMIT_SIGN, true));
```

Comment:
@olamy This is actually a tri-state where `true` means force signing (which often fails with `Signing service not available`). I think the default should either be `null` or `false`....

Reply:
Indicates that a JGit test case is somehow missing here as well...

Reply:
Would be great.
Iintiallly,  I added this option because I have this in `~/.gitconfig` and so couldn't run unit test without failures.

```
[commit]
        gpgSign = true
[tag]
        gpgSign = true
```